src/components/Dropdowns/AccessibleDropdown.tsx

```diff
@@ -0,0 +1,242 @@
+   *
+   * @param props: The props passed to the dropdown component.
+   * @param isOpen: Whether the dropdown is currently open.
+   * @param onClick: The callback to open the dropdown.
+   *
+   * Overrides must call the provided `onClick` callback during click or interaction events for the
+   * dropdown to work correctly.
+   *
+   * Tooltips may not work as expected if the root element is not a button or does not take callbacks
+   *  as props for the following events: `onMouseEnter/Leave`, `onPointerEnter/Leave`,
+   * `onFocus`, or `onBlur`. See implementation of `IconButton.tsx` for an example.
+   */
+  renderButton?: ((props: AccessibleDropdownProps, isOpen: boolean, onClick: () => void) => ReactElement) | null;
+  onButtonClicked?: (key: string) => void;
+  /** Whether the tooltip should appear when hovered. */
+  showTooltip?: boolean;
+  /** If null, uses button text. */
+  tooltipText?: string | null;
+  /** Width of the dropdown. Overrides the default sizing behavior if set. */
+};
+
+const defaultProps: Partial<AccessibleDropdownProps> = {
+  label: null,
+  renderDropdownContent: null,
+  disabled: false,
+  buttonType: "outlined",
+  showTooltip: true,
+  tooltipText: null,
+  buttonStyle: {},
+  renderButton: null,
+  onButtonClicked: () => {},
+};
+
+/** Contains the main dropdown button and label text */
+const MainContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  width: max-content;
+  gap: 6px;
+`;
+
+/** Button that is clicked to open the dropdown */
+const MainButton = styled(Button)<{ $open: boolean; $type: AccessibleDropdownProps["buttonType"] }>`
+  width: 15vw;
+  // Override Antd width transition
+  transition: all 0.2s cubic-bezier(0.645, 0.045, 0.355, 1), width 0s;
+
+  // Override Ant styling for the outlined button style
+  ${(props) => {
+    if (props.$type === "outlined") {
+      return css`
+        &.ant-btn:not(:disabled) {
+          border-color: var(--color-borders);
+          color: var(--color-text);
+        }
+
+        &.ant-btn:not(:disabled):hover {
+          background-color: transparent;
+          border-color: var(--color-button);
+          color: var(--color-text); // Repeated to override color changes
+        }
+
+        &.ant-btn:not(:disabled):active {
+          background-color: transparent;
+          border-color: var(--color-button-active);
+          color: var(--color-text);
+        }
+      `;
+    }
+    return;
+  }}
+
+  // When the modal is opened ("pinned") by clicking on it, show an
+  // extra active-style outline.
+  ${(props) => {
+    if (props.$open) {
+      return css`
+        &:not(:disabled) {
+          border-color: var(--color-button-active) !important;
+        }
+      `;
+    }
+    return;
+  }}
+`;
+
+/** Container for the text label and dropdown arrow SVG inside the main dropdown-triggering button. */
+const MainButtonContents = styled.div`
+  width: 100%;
```

Comment:
I originally thought this would help with the `justify-content: space-between;` attribute by allowing the contents to take up the full width, but agreed, it looks like this isn't doing anything. The `width: 100%` in the div selector below does prevent some strange formatting bugs though.